dapp/src/components/TransactionHistory.js

```diff
@@ -184,63 +183,28 @@ const TransactionHistory = ({ isMobile }) => {
   const setFilters = (filters) => {
     _setFilters(filters)
     setCurrentPage(1)
+  }
+
+  const historyQuery = useTransactionHistoryQuery(account)
```

Comment:
This hook can actually be used even multiple times in a page and `react-query` is smart enough to dedupe the requests into one!

(so imagine you got multiple sections in the app relying on the history data, you can just use this hook and it won't make an extra network request)

Reply:
nice improvement, thanks